src/RefreshOptions.ts

```diff
@@ -24,6 +24,11 @@ export interface RefreshOptions {
      * Any modifications to watched settings will refresh all settings loaded by the configuration provider when refresh() is called.
      */
     watchedSettings?: WatchedSetting[];
+
+    /**
+     * Specifies whether all configuration settings will be watched for changes on the server.
+     */
+    watchAll?: boolean;

```

Comment:
I prefer `watchAll` tp `registerAll`. In .NET, we have `RefreshOptions.Register` method. But, in javascript, we let user to list watchedSettings in the refreshOption.

Reply:
Please don't follow the .NET design. When RefreshOptions.enabled is true but no individual keys are watched, it's watching all.

Reply:
Just like .NET provider, if sentinel key is not provided, JS provider will also throw an error today. Are we saying that we stop throwing that error and assume that no sentinel means all selected kv are watched?

Reply:
Note that .NET doesn't have an explicit option to enable refresh. The Register* API serves two purposes: 1. enable refresh 2. answer what to monitor. This is not the case with other providers. We have an explicit option to enable refresh in all other providers. This is why we don't need anything else in other providers to monitoring all.